frontend/modelarchive/src/test/java/org/pytorch/serve/archive/ModelArchiveTest.java

```diff
@@ -21,48 +25,70 @@ public void beforeTest() {
         FileUtils.deleteQuietly(new File(tmp, "models"));
     }
 
-    @Test(expectedExceptions = ModelNotFoundException.class)
+    @Test
     public void test() throws ModelException, IOException {
         String modelStore = "src/test/resources/models";
-
-        ModelArchive archive = ModelArchive.downloadModel(modelStore, "noop.mar");
+        ModelArchive archive = ModelArchive.downloadModel(VALID_HOSTS_LIST, modelStore, "noop.mar");
         archive.validate();
         archive.clean();
         Assert.assertEquals(archive.getModelName(), "noop");
-
-        // load model for s3 --> This will fail as this model is not compatible with
-        // new implementation.
-        // TODO Change this once we have example models on s3
-        archive =
-                ModelArchive.downloadModel(
-                        modelStore,
-                        "https://s3.amazonaws.com/model-server/models/squeezenet_v1.1/squeezenet_v1.1.model");
-        Assert.assertEquals(archive.getModelName(), null);
-        ModelArchive.removeModel(
-                modelStore,
-                "https://s3.amazonaws.com/model-server/models/squeezenet_v1.1/squeezenet_v1.1.model");
-        Assert.assertTrue(!new File(modelStore, "squeezenet_v1.1.model").exists());
-        ModelArchive.downloadModel(modelStore, "/../noop-v1.0");
     }
 
     @Test(expectedExceptions = DownloadModelException.class)
     public void testInvalidURL() throws ModelException, IOException {
         String modelStore = "src/test/resources/models";
-        // load model for s3 --> This will fail as this model is not compatible with
-        // new implementation.
-        // TODO Change this once we have example models on s3
         ModelArchive.downloadModel(
+                VALID_HOSTS_LIST,
                 modelStore,
                 "https://s3.amazonaws.com/model-server/models/squeezenet_v1.1/squeezenet_v1.1.mod");
     }
 
     @Test(expectedExceptions = DownloadModelException.class)
     public void testMalformURL() throws ModelException, IOException {
         String modelStore = "src/test/resources/models";
-        // load model for s3 --> This will fail as this model is not compatible with
-        // new implementation.
-        // TODO Change this once we have example models on s3
         ModelArchive.downloadModel(
-                modelStore, "https://../model-server/models/squeezenet_v1.1/squeezenet_v1.1.mod");
+                VALID_HOSTS_LIST,
+                modelStore,
+                "https://../model-server/models/squeezenet_v1.1/squeezenet_v1.1.mod");
+    }
+
+    @Test(expectedExceptions = ModelNotFoundException.class)
+    public void testValidHostURL() throws ModelException, IOException {
```

Comment:
@shivamshriwas Please add another test for multiple urls test scenario, and failure cases where url is an invalid one

Reply:
- Added multiple urls test scenario
- enhanced existing test cases as well with now exception messages also being compared.